nvaccess/nvda
# NVDAObject.presentationType: groupings with position information but no name and no description should still be treated as content I.e. announced in focus ancestry.

### Link to issue number:
None

### Summary of the issue:
NVDA filteres controls with a role of groupping out of being presented in the focus ancestry if they have no name or description. However, there is a trend with some newer Microsoft products to add position info to these unlabeled groupings (E.g. grouping 1 of 5) and it is requested that NvDA announce these as they provide very deliberate topology to the user.
 
### Description of user facing changes
NVDA will now report unlabeled groupings that have useful position information, such as in recent versions of Microsoft Office 365 menus.
 
### Description of development approach
NVDA.presentationType now treats groupings specifically, only changing them to layout if they have no name, no description, and no position information.

### Testing strategy:
* Open a recent build of Microsoft Word 365.
* Type a spelling mistake.
* Open the context menu on the mistake, and arrow up and down the menu. Notice that the suggestions are in "grouping 1 of 2" and the Add to Dictionary and Ignore items are in "grouping 2 of 2".
Note: Some builds of word 365 have started placing a whatespace character in the nameof unlabeld groupings to force NVDA to get this behaviour without this pr. but this was temporary. thus when testing this, strip whitespace from UIA NVDAObject's name property first.
 
### Known issues with pull request:

### Change log entries:
New features
Changes
Bug fixes
NVDA will now report unlabeled groupings that have useful position information, such as in recent versions of Microsoft Office 365 menus.
For Developers

### Code Review Checklist:



  - description is up to date
  - change log entries
  - Unit tests
  - System (end to end) tests
  - Manual testing
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
